dotcom-rendering/.storybook/decorators/grid.ts

```diff
@@ -0,0 +1,164 @@
+// ----- Imports ----- //
+
+import { from } from '@guardian/source-foundations';
+
+// ----- Columns & Lines ----- //
+
+/**
+ * Named CSS grid lines, based on the three columns commonly used for Guardian
+ * layouts.
+ */
+type Line =
+	| 'viewport-start'
+	| 'left-column-start'
+	| 'left-column-end'
+	| 'centre-column-start'
+	| 'centre-column-end'
+	| 'right-column-start'
+	| 'right-column-end'
+	| 'viewport-end';
+
+const mobileColumns =
+	'[viewport-start] 0px [centre-column-start] repeat(4, 1fr) [centre-column-end] 0px [viewport-end]';
+const tabletColumns =
+	'[viewport-start] 1fr [centre-column-start] repeat(12, 40px) [centre-column-end] 1fr [viewport-end]';
+const desktopColumns =
+	'[viewport-start] 1fr [centre-column-start] repeat(8, 60px) [centre-column-end right-column-start] repeat(4, 60px) [right-column-end] 1fr [viewport-end]';
+const leftColColumns =
+	'[viewport-start] 1fr [left-column-start] repeat(2, 60px) [left-column-end centre-column-start] repeat(8, 60px) [centre-column-end right-column-start] repeat(4, 60px) [right-column-end] 1fr [viewport-end]';
+const wideColumns =
+	'[viewport-start] 1fr [left-column-start] repeat(3, 60px) [left-column-end centre-column-start] repeat(8, 60px) [centre-column-end] 60px [right-column-start] repeat(4, 60px) [right-column-end] 1fr [viewport-end]';
+const mobileColumnGap = '10px';
+const columnGap = '20px';
```

Comment:
Is there an issue in keeping this in sync with the grid used by the real layouts?

Reply:
Yes. There are several grids used by the full pages, because they're often defined per-layout. That's the reason I didn't try to reuse them for this, but it does mean we've got multiple grid definitions in use, even without this change.

I'd like to consider whether we can unify them, since I'd expect the rules to generally be the same. As Max says, we've been using this grid design for a long time.